### PR TITLE
New chart major version

### LINKED ISSUE #679: Helm Chart 0.17.0
Hi,

Could you publish a new helm chart with version 0.17.0 ?

Thanks

### PR DESCRIPTION
**Pull request by juan131**
Signed-off-by: juan131 <juanariza@vmware.com>

**Description of the change**

These changes perform a major refactorization of the SealedSecrets Helm chart so it adopts several common practices and feature available in Bitnami charts.

The PR also updates the SealedSecrets version so it uses the latest available version: `0.17.1`

**Benefits**

- Standardization
- Compatible with new objects api versions.

**Possible drawbacks**

Users will have to adapt their current values.yaml (or parameters passed via `--set`) to the new values structure.

**Applicable issues**

- fixes #679

**Additional information**

This PR also adapts the values.yaml to be compatible with the [readme-generator tool](https://github.com/bitnami-labs/readme-generator-for-helm).

### CODE REVIEW

**Review by agarcia-oss — APPROVED** [review]
Amazing work, @juan131 👏 👏 👏

File: helm/sealed-secrets/templates/role-binding.yaml [hunk 1]
```diff
@@ -2,45 +2,35 @@
...
 metadata:
-  name: {{ template "sealed-secrets.fullname" . }}-key-admin
-  namespace: {{ template "sealed-secrets.namespace" . }}
-  labels:
-    app.kubernetes.io/name: {{ template "sealed-secrets.name" . }}
-    helm.sh/chart: {{ template "sealed-secrets.chart" . }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/version: {{ .Chart.AppVersion }}
+  name: {{ include "sealed-secrets.fullname" . }}-key-admin
```
Comment (alvneiayu) [inline helm/sealed-secrets/templates/role-binding.yaml:5]: Suggestion: Use printf here?

File: helm/sealed-secrets/templates/role-binding.yaml [hunk 2]
```diff
@@ -2,45 +2,35 @@
...
 metadata:
-  name: {{ template "sealed-secrets.fullname" . }}-service-proxier
-  namespace: {{ template "sealed-secrets.namespace" . }}
-  labels:
-    app.kubernetes.io/name: {{ template "sealed-secrets.name" . }}
-    helm.sh/chart: {{ template "sealed-secrets.chart" . }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/version: {{ .Chart.AppVersion }}
+  name: {{ include "sealed-secrets.fullname" . }}-service-proxier
```
Comment (alvneiayu) [inline helm/sealed-secrets/templates/role-binding.yaml:24]: Ditto (printf)

File: helm/sealed-secrets/templates/role-binding.yaml [hunk 3]
```diff
@@ -2,45 +2,35 @@
...
+  labels: {{- include "sealed-secrets.labels" . | nindent 4 }}
     {{- if .Values.rbac.labels }}
-{{ toYaml .Values.rbac.labels | indent 4 }}
+    {{- toYaml .Values.rbac.labels | nindent 4 }}
     {{- end }}
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: Role
-  name: {{ template "sealed-secrets.fullname" . }}-service-proxier
+  name: {{ include "sealed-secrets.fullname" . }}-service-proxier
```
Comment (alvneiayu) [inline helm/sealed-secrets/templates/role-binding.yaml:33]: Ditto (printf)

File: helm/sealed-secrets/templates/role.yaml [hunk 4]
```diff
@@ -2,16 +2,11 @@
...
 metadata:
-  name: {{ template "sealed-secrets.fullname" . }}-key-admin
-  namespace: {{ template "sealed-secrets.namespace" . }}
-  labels:
-    app.kubernetes.io/name: {{ template "sealed-secrets.name" . }}
-    helm.sh/chart: {{ template "sealed-secrets.chart" . }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/version: {{ .Chart.AppVersion }}
+  name: {{ include "sealed-secrets.fullname" . }}-key-admin
```
Comment (alvneiayu) [inline helm/sealed-secrets/templates/role.yaml:5]: Ditto (printf)

File: helm/sealed-secrets/templates/role.yaml [hunk 5]
```diff
@@ -33,23 +28,18 @@ rules:
...
 metadata:
-  name: {{ template "sealed-secrets.fullname" . }}-service-proxier
-  namespace: {{ template "sealed-secrets.namespace" . }}
-  labels:
-    app.kubernetes.io/name: {{ template "sealed-secrets.name" . }}
-    helm.sh/chart: {{ template "sealed-secrets.chart" . }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/version: {{ .Chart.AppVersion }}
+  name: {{ include "sealed-secrets.fullname" . }}-service-proxier
```
Comment (alvneiayu) [inline helm/sealed-secrets/templates/role.yaml:31]: Ditto (printf)

**Review by alvneiayu — APPROVED** [review]
Awesome job @juan131. I included some suggestions but if you want to skip them it is ok. LGTM

**Pull request merged**